.build/config/jobs.cfg

```diff
@@ -0,0 +1,183 @@
+# limitations under the License.
+
+#
+# Contains definitions of all ci jobs in the system. Composed into pipelines; see pipelines.cfg
+#
+
+# Job config format is key-value, colon delimited.
+#   Order does not matter, and subsequent jobs are delimited by either subsequent "job:N" tags or EOF
+#   All parameters should be processed in sequential order with child parameters executing after parent
+#
+#   job:<name>
+#   description:<description>
+#   env:{key=val,key=val,key=val}
+#       string:Comma-delimited list of key value pairs to export in runtime env for test
+#       NOTE: env parameters should *all* be evaluated from the root of the hierarchy. Child env params that
+#               override the parent are acceptable but the totality of env should be a composite of each env:
+#               from root on down in sequential order.
+#   cmd:<to_run>
+#       string:Command used to run the tests resolved by testlist
+#   testlist:{command}
+#       string:Command run on system to generate comma-delimited list of all test units to run in target
+#           Note:This is expected to run in a terminal shell context
+#   The following parameters indicate the max allowable of a given resource
+#       memory:N
+#           integer:in GB
+#       cpu:N
+#           integer:logical cpu count
+#       storage:N
+#           integer:GB allowable used
+#
+# A job can inherit the configuration from another job via the following syntax:
+#   job:base_job
+#       ... params ...
+#   job:child
+#       parent:base_job
+#       override:new_val
+# Any parameter defined in the child job is expected to override the parameters of the parent
+#
+# Other parameters of interest:
+#   Test timeouts per suite are in build.xml
+
+#-----------------------------------------------------------------------------
+# Base reference profile; add new parameters here to ensure all jobs have a reasonable default
+#-----------------------------------------------------------------------------
+job:base
+    memory:6
+    cpu:4
+    storage:100
+    env:ANT_TEST_OPTS="-Dno-build-test=true -Dtmp.dir=${TMP_DIR}"
+
+#-----------------------------------------------------------------------------
+# JVM based single node tests
+#-----------------------------------------------------------------------------
+job:unit
+    parent:base
+    testlist:find test/unit -name '*Test.java' | sed "s;^test/unit/;;g" | sort
+    cmd:ant test ${ANT_TEST_OPTS}
+
+job:cdc
+    parent:unit
+    description:Runs test suites with CDC enabled
+
+job:compression
+    parent:unit
+    description:Runs test suites with SSTable compression enabled
+
+job:test-oa
+    parent:unit
+    description:Tests oa sstable format
+
+job:test-system-keyspace-directory
+    parent:unit
+    description:Execute unit tests with a system keyspaces directory configured
+
+job:test-tries
+    parent:unit
+    description:Execute unit tests with trie memtables
```

Comment:
I strongly feel we should not define each like this and instead should push this into the oringal `unit` definition

```
job: unit
  parent:base
  test_cases:
    - cdc: [true, false]
    - compression: [true, false]
    - oa: [true, false]
    - system_keyspace_directory: [true, false]
    - tries: [true, false]
```

doing that is clear that these are just running `unit` with different configs.  Now the one question becomes how we "mix"... should tries run with compression enabled and cdc enabled?  Should we base this on the environment that can handle such cases?

Reply:
Hm. It'd certainly make things read more cleanly.

The parameterization bit of it and encoding of positional parameters within a config file format concerns me. There's no locality between the bool's here and what they represent, and then you end up with downstream parsers that also need to couple w/that positionality rather than being able to retain a declarative relationship between key/value.